core/controllers/acl_decorators.py

```diff
@@ -2318,11 +2338,47 @@ def test_can_access(self, topic_name, **kwargs):
         Raises:
             PageNotFoundException: The given page cannot be found.
         """
-        topic_name = python_utils.url_unquote_plus(topic_name)
-        topic = topic_fetchers.get_topic_by_name(topic_name)
+        lowercase_abbrev_topic_name = (
+            python_utils.url_unquote_plus(abbrev_topic_name).replace('-', ' '))
+        topic = topic_fetchers.get_topic_by_abbreviated_name(
+            lowercase_abbrev_topic_name)
+        classroom_name = kwargs.pop('classroom_name', None)
```

Comment:
If this is an expected argument, why not pass it in explicitly?

Reply:
Then use two handlers, or have the parameter be required (but unused) even for the data handler. Don't do things implicitly like this.